packages/playwright-core/src/client/channelOwner.ts

```diff
@@ -188,7 +189,7 @@ export abstract class ChannelOwner<T extends channels.Channel = channels.Channel
       logApiCall(logger, `=> ${apiName} started`, isInternal);
       const apiZone: ApiZone = { apiName, frames, isInternal, reported: false, csi, callCookie, wallTime };
       const result = await zones.run('apiZone', apiZone, async () => await func(apiZone));
-      csi?.onApiCallEnd(callCookie);
+      await csi?.onApiCallEnd(callCookie);
```

Comment:
similar issue happens here when this is not awaited - the test runs to completion, closes the browser and code in `onApiCallEnd` can get rejected with "Test ended." error because the browser was already closes by the time it gets executed